### PR TITLE
Reqwest example

### PR DESCRIPTION
**Pull request by unleashed**
This is an example using the library with Reqwest. Preferably merge after #36.

To run this example you'll need to:

> $ cargo run --example reqwest-report --features reqwest-sync

Alternatively you can pass --all-features instead of the exact features required by the example.

### CODE REVIEW

File: examples/reqwest-report.rs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        "userkey_3",
+        "userkey 4",
+        "userkey 5",
+    ];
+    let apps = uks
+        .into_iter()
+        .map(|uk| Application::from(UserKey::from(*uk)))
+        .collect::<Vec<_>>();
+
+    println!("Apps: {:#?}", apps);
```
Comment (davidor) [inline examples/reqwest-report.rs]: The output of this seems to have an extra `UserKey`.

```
Apps: [                                                                                                                                                                         
    UserKey(                                                                                                                                                                    
        UserKey(                                                                                                                                                                
            "userkey_1",                                                                                                                                                        
        ),                                                                                                                                                                      
    ),                                                                                                                                                                          
    UserKey(                                                                                                                                                                    
        UserKey(                                                                                                                                                                
            "userkey_2",                                                                                                                                                        
        ),                                                                                                                                                                      
    ), 
```

File: examples/reqwest-report.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    let txns = apps
+        .iter()
+        .zip(&usages)
+        .map(|(a, u)| Transaction::new(a, None, Some(u), Some(&ts)))
+        .collect::<Vec<_>>();
+
+    let mut extensions = Extensions::new();
+    extensions.insert("no_body", "1");
+    extensions.insert("testing[=]", "0[=:=]0");
```
Comment (davidor) [inline examples/reqwest-report.rs]: ?

Comment (unleashed) [reply examples/reqwest-report.rs]: this allows you to see how the escaping is done

File: examples/reqwest-report.rs (hunk 1 above)
Comment (unleashed) [reply examples/reqwest-report.rs]: this is correct, it is a debug view of `Application::UserKey(UserKey)`

Comment (unleashed) [reply examples/reqwest-report.rs]: (perhaps we might want to look into whether we'd want to change/simplify it in a different PR)

Comment (davidor) [reply examples/reqwest-report.rs]: Yes, we should :+1:

**Comment by unleashed** [comment]
bors r=@davidor p=5

**Comment by unleashed** [comment]
bors r=@davidor,@PhilipGough

**Pull request merged**